src/gsumm.c

```diff
@@ -761,28 +761,62 @@ SEXP gmin(SEXP x, SEXP narm)
     break;
   case REALSXP:
     ans = PROTECT(allocVector(REALSXP, ngrp)); protecti++;
-    if (!LOGICAL(narm)[0]) {
-      for (i=0; i<ngrp; i++) REAL(ans)[i] = R_PosInf;
-      for (i=0; i<n; i++) {
-        thisgrp = grp[i];
-        ix = (irowslen == -1) ? i : irows[i]-1;
-        if (ISNAN(REAL(x)[ix]) || REAL(x)[ix] < REAL(ans)[thisgrp])
-          REAL(ans)[thisgrp] = REAL(x)[ix];
-      }
-    } else {
-      for (i=0; i<ngrp; i++) REAL(ans)[i] = NA_REAL;
-      for (i=0; i<n; i++) {
-        thisgrp = grp[i];
-        ix = (irowslen == -1) ? i : irows[i]-1;
-        if (ISNAN(REAL(x)[ix])) continue;
-        if (ISNAN(REAL(ans)[thisgrp]) || REAL(x)[ix] < REAL(ans)[thisgrp])
-          REAL(ans)[thisgrp] = REAL(x)[ix];
+    if (INHERITS(x, char_integer64)) {
+      int64_t *ansp = (int64_t *)REAL(ans);
+      int64_t *xp = (int64_t *)REAL(x);
+      if (LOGICAL(narm)[0]) {
+        for (i=0; i<ngrp; i++) ansp[i] = NA_INTEGER64;
+        for (i=0; i<n; i++) {
+          thisgrp = grp[i];
+          ix = (irowslen == -1) ? i : irows[i]-1;
+          if (xp[ix] == NA_INTEGER64) continue;
+          if (ansp[thisgrp] == NA_INTEGER64 || xp[ix] < ansp[thisgrp])
+            ansp[thisgrp] = xp[ix];
+        }
+        for (i=0; i<ngrp; i++) {
+          if (ansp[i] == NA_INTEGER64) {
+            warning(_("No non-missing values found in at least one group. Returning 'MAX_INTEGER64' for such groups to be consistent with bit64::max.integer64"));
```

Comment:
Maybe better to do `Returning '%d'` and substitute `MAX_INTEGER64` but I can never remember the format code for this input type...

Reply:
nvm, done